crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -614,25 +614,24 @@ where
     /// Returns `Poll::Ready(Ok(()))` when no buffered items remain.
     fn poll_flush(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
         let mut this = self.project();
-        loop {
-            match ready!(this.inner.as_mut().poll_flush(cx)) {
-                Err(err) => {
-                    trace!(target: "net::p2p",
-                        %err,
-                        "error flushing p2p stream"
-                    );
-                    return Poll::Ready(Err(err.into()))
-                }
-                Ok(()) => {
+        let poll_res = loop {
+            match this.inner.as_mut().poll_ready(cx) {
```

Comment:
this loop is actually redundant.

as described in the issue I'd like to solve this differently, because this currently also doesn't really make use of the framed buffer, so I don't really want to revert this again but improve it instead

Reply:
I don't see how it doesn't make use of the framed buffer?

Reply:
`Framed` wraps `FramedImpl`

https://docs.rs/tokio-util/latest/src/tokio_util/codec/framed.rs.html#17-41

this is how `FramedImpl` implements `poll_ready`, it flushes if enough messages are buffered to fill a frame

https://docs.rs/tokio-util/latest/src/tokio_util/codec/framed_impl.rs.html#261-267

`poll_flush` impl for `P2PStream`, calls `poll_flush` on framed explicitly before returning to make sure all messages are sent. this last frame may be partly full.